modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/upgrade/v_1_2_0/UpgradeStagingGroupTypeSettings.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.calendar.web.internal.upgrade.v_1_2_0;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Balázs Sáfrány-Kovalik
+ */
+public class UpgradeStagingGroupTypeSettings extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_removeStagedCalendarInstanceId();
+	}
+
+	private String _getNewTypeSettings(String typeSettings) {
+		Pattern pattern = Pattern.compile(
+			_STAGED_CALENDAR_PORTLET + _INSTANCE_SEPARATOR + "\\w{12}",
+			Pattern.MULTILINE);
+
+		Matcher matcher = pattern.matcher(typeSettings);
+
+		return matcher.replaceAll(_STAGED_CALENDAR_PORTLET);
+	}
+
+	private void _removeStagedCalendarInstanceId() throws Exception {
```

Comment:
Hi @balazssk,

What we need to analyze is where the upgrade process adds the sufix _INSTANCE_XXX to the calendar name in the typeSettings for the group record, w ehave to fix that. During my tests before starting the upgrade there wasn't an INSTANCE_ sufix in that field.

Reply:
yes, if I remember correctly it gets added as the Calendar portlet uses the instanceable upgrade from BaseUpgradePortletId while the other portlets are using the uninstanceable.